src/node_js/closures.rs

```diff
@@ -0,0 +1,461 @@
+            //             dbg!(::napi::sys::napi_create_reference(
+            //                 env.raw(),
+            //                 func.raw(),
+            //                 1,
+            //                 &mut raw_ref,
+            //             ));
+            //             (func, raw_ref, env.raw())
+            //         },
+            //         // drop: call unref
+            //         |(_, raw_ref, raw_env)| unsafe {
+            //             // dbg!(::napi::sys::napi_reference_unref(
+            //             //     raw_env, raw_ref, &mut 0,
+            //             // ));
+            //             // dbg!(::napi::sys::napi_delete_reference(
+            //             //     raw_env, raw_ref,
+            //             // ));
+            //         },
+            //     );
+
+            //     let func = UnsafeAssertSendSync(func);
+            //     struct UnsafeAssertSendSync<T>(T);
+            //     unsafe impl<T> Send for UnsafeAssertSendSync<T> {}
+            //     unsafe impl<T> Sync for UnsafeAssertSendSync<T> {}
+            //     let wrapped_func = env.create_function_from_closure("<name>", move |ctx: CallContext<'_>| Ok({
+            //         drop(&func);
+            //         let _ =
+            //             ret_sender.send(
+            //                 (unsafe {
+            //                     JsFunction::from_raw_unchecked(
+            //                         ctx.env.raw(),
+            //                         func.0.0.raw(),
+            //                     )
+            //                     // func.0.0
+            //                 }).call(
+            //                     ctx.this::<JsObject>().ok().as_ref(),
+            //                     &ctx.get_all(),
+            //                 )
+            //                 .and_then(|js_unknown| {
+            //                     use ::core::convert::TryInto;
+            //                     js_unknown
+            //                         .try_into()
+            //                         .map_or_else(
+            //                             |_| todo!(),
+            //                             |napi_val| ReprNapi::from_napi_value(ctx.env, napi_val),
+            //                         )
+            //                 })
+            //             )
+            //         ;
+            //         ctx .env
+            //             .get_undefined()
+            //             .unwrap()
+            //     }))?;
+            //     (wrapped_func, ret_receiver)
+            // };
+
+            let mut ts_fun = ThreadsafeFunction::create(
+                env.raw(),
+                fun.get_thread_tied_func().unwrap(),
+                // max_queue_size /* use `0` for a sane default */
+                1,
+                // callback
+                Self::handle_params,
+            )?;
+            ts_fun.refer(&env)?; // Keep the main loop alive while this entity is.
```

Comment:
The comment... does it mean "keep the main loop alive _while this entity is alive_", or is that "is" at the end about something else?

Reply:
True, I've clarified that comment 🙂